tensorflow/core/kernels/linalg/matrix_diag_op.cc

```diff
@@ -236,13 +236,21 @@ class MatrixDiagOp : public OpKernel {
         errors::InvalidArgument(
             "lower_diag_index must not be larger than upper_diag_index: ",
             lower_diag_index, " > ", upper_diag_index));
-    OP_REQUIRES(context,
-                lower_diag_index == upper_diag_index ||
-                    diagonal_shape.dim_size(diag_rank - 2) == num_diags,
```

Comment:
`max(diag_rank - 2, 0)`?

Reply:
> `max(diag_rank - 2, 0)`?

Oh.You mean instead of validating for diag_rank==1 and diag_rank>1 separately, we can do it in simple way by taking  `max(diag_rank - 2, 0)`. It seems better and avoid redundancy. 

May be this check works now. I am assuming `max` should be from `std` ?

```
OP_REQUIRES(context,
    lower_diag_index == upper_diag_index ||
        diagonal_shape.dim_size(std::max(diag_rank - 2, 0)) == num_diags,
    errors::InvalidArgument(
        "The number of diagonals provided in the input does not "
        "match the lower_diag_index and upper_diag_index range."));
```

Reply:
yes